GirafRest.Test/Controllers/UserControllerTest.cs

```diff
@@ -458,9 +507,15 @@ public void GetCitizens_AuthenticationChecks(int authUser, int userToEdit, Error
             var usercontroller = initializeTest();
             var user = _testContext.MockUsers[authUser];
             _testContext.MockUserManager.MockLoginAsUser(user);
-            var res = usercontroller.GetCitizens(_testContext.MockUsers[userToEdit].Id).Result;
-
-            Assert.Equal(expectedError, res.ErrorCode);
+            var res = usercontroller.GetCitizens(_testContext.MockUsers[userToEdit].Id).Result as ObjectResult;
+
+            if (expectedError != ErrorCode.NoError) {

```

Comment:
This does not seem like the correct way to do a unit test, instead; have two tests, one that tests success and one that tests errors, that will make it much easier to see why either fails later